src/v/debug_bundle/debug_bundle_service.cc

```diff
@@ -338,9 +338,18 @@ ss::future<result<void>> service::initiate_rpk_debug_bundle_collection(
 
     // Kick off the wait by waiting for the process to finish and then emplacing
     // the result
-    ssx::spawn_with_gate(_gate, [this]() {
+    ssx::spawn_with_gate(_gate, [this, job_id]() {
         return _rpk_process->wait()
-          .then([this](auto res) { _rpk_process->emplace_wait_result(res); })
+          .then([this, job_id](auto res) {
+              // THis ensures in the extremely unlikely case where this
+              // continuation is called after another debug bundle has been
+              // initiated, that we are accessing a present and valid
```

Comment:
Ahh, this can be cleaned up with my previous suggestion of setting `_wait_result` inside `wait`.